### PR TITLE
added validation scripts and gitignores

### PR DESCRIPTION
**Pull request by swathiswaminathan**
- Added scripts for make create, make validate and make teardown for initial gke-tf cluster setup.
- Added Jenkinsfile
- updated makefile
- added more items in gitignore

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,113 @@
+#!/usr/bin/env groovy
```
Comment (chrislovecnm) [inline Jenkinsfile]: @bgeesaman Which project has the best Jenkinsfile now??

File: scripts/setup-terraform-test.sh [hunk 2]
```diff
@@ -32,17 +32,22 @@ fi
...
-do
-   :
-	# Initialize and run Terraform
-	echo "Setting up test for ${i}"
-	echo "Running terraform init and terraform plan"
-	(cd "${i}"; terraform init -input=false)
-	(cd "${i}"; terraform plan -input=false -var-file="$ROOT/terraform.tfvars")
-done
+# export PROJECT="$(gcloud config list project --format='value(core.project)')"
+gke-tf gen -d ./terraform -f gke-tf-demo.yaml -o -p ${PROJECT}
```
Comment (chrislovecnm) [inline scripts/setup-terraform-test.sh]: We are going to have to d/l gke-tf or install it in a Jenkins container.  Also the gke-tf binary needs to match the OS that this script is running on.

File: Jenkinsfile (hunk 1 above)
Comment (bgeesaman) [reply Jenkinsfile]: Vault has a clean/simple/updated one now.

File: scripts/setup-terraform-test.sh (hunk 2 above)
Comment (swathiswaminathan) [reply scripts/setup-terraform-test.sh]: Hello @chrislovecnm - We're downloading the gke-tf, instructions for downloading it are pulled from common.sh script.

File: Jenkinsfile (hunk 1 above)
Comment (swathiswaminathan) [reply Jenkinsfile]: Thanks @chrislovecnm / @bgeesaman - added jenkinsfile from vault

File: scripts/common.sh [hunk 3]
```diff
@@ -19,21 +19,85 @@
...
+
+  # Download a known version of gke-tf
+  GKE_TF_PATH="${ROOT}/bin/gke-tf"
+
+  # If gke-tf is not already installed and executable
+  if ! [ -x "${GKE_TF_PATH}" ]; then
+
+    # Install a version of gke-tf to ${ROOT}/bin/gke-tf
+    # CI runs in a Linux environment
+    # curl -sLO "https://github.com/GoogleCloudPlatform/gke-terraform-generator/releases/download/0.1-alpha.1/gke-tf-darwin-amd64"
```
Comment (chrislovecnm) [inline scripts/common.sh]: You can check the OS version with a call to `uname`.  For example on my Mac.  I think uname on linux 

```console
$ uname
Darwin
```

**Comment by chrislovecnm** [comment]
test this please

**Comment by wripley** [comment]
test this please

**Pull request merged**